### ISSUE TITLE
Can bolt resolve non-forge dependencies in the ``metadata.json``?

### ISSUE THREAD

**Issue by gavindidrichsen** [puppetlabs/bolt#3276]
## Use Case

I would like Bolt to bring dependent non-Forge modules w/o specifying them directly in the Bolt project.

## Describe the Solution You Would Like

For example, the[ puppetlabs-chocolatey_server metadata.json](https://github.com/chocolatey-community/puppet-chocolatey_server/blob/6f72c3554306f6baf264fc599af92dbe89a54595/metadata.json), contains the following forge dependencies:

```json
  "dependencies": [
    {
      "name": "puppetlabs/acl",
      "version_requirement": ">= 1.0.1 < 3.0.0"
    },
    {
      "name": "puppetlabs/powershell",
      "version_requirement": ">= 1.0.4 < 3.0.0"
    },
    {
      "name": "puppetlabs/iis",
      "version_requirement": ">= 4.0.0 < 5.0.0"
    }
```

I'd like to be able to add non-forge modules to the dependency list above, if possible.  This could be, for example, a ``git@gitub.com`` or an ``https://github.com/org/myrepo.git``.

## Describe Alternatives You've Considered
N/A

## Additional Context

This [voxpupuli discussion](https://github.com/orgs/voxpupuli/discussions/7#discussioncomment-8695059) goes into more detail about the actual ask:

> How can I bring that dependency to my main project without specifying it explicitly in the bolt-project.yaml? Ideally, I'd like to specify the my_basemodule as a dependency in the my_module1's metadata.json file. Then I'd expect that bolt module install --force will bring it to my project. Is it possible?

In other words,

> I would like Bolt to bring dependent non-Forge modules w/o specifying them directly in the Bolt project.

**Comment by donoghuc** [comment]
This is not currently possible in bolt's dependency resolution for modules it is noted in the docs: https://www.puppet.com/docs/bolt/latest/bolt_installing_modules.html#add-a-forge-module-and-its-dependencies-to-an-existing-project 

As mentioned these dependencies (non forge hosted) will need to be listed in the projects module deps.
